### PR TITLE
Added a working AdaptiveTrigger implementing ITriggerValue

### PR DESCRIPTION
**Pull request by LocalJoost**
Added a working AdaptiveTrigger implementing ITriggerValue so it can be used in with CompositeStateTrigger

### CODE REVIEW

File: src/TestApp/MainPage.xaml [hunk 1]
```diff
@@ -7,75 +7,79 @@
...
-                    <samples:FullScreenSample />
-                </local:SampleContainer>
-                
-                <local:SampleContainer Title="HostedTrigger">
-                    <samples:IsHostedSample />
-                </local:SampleContainer>
-            </VariableSizedWrapGrid>
-		</ScrollViewer>
-	</Grid>
+  <Grid Background="DimGray">
```
Comment (dotMorten) [inline src/TestApp/MainPage.xaml:10]: Please fix the indentation so this doesn't show as a change

File: src/WindowsStateTriggers.sln [hunk 2]
```diff
@@ -47,6 +47,8 @@ Global
 		{52D52F3C-4F64-4957-9FE8-780B7D7F75C3}.Release|x86.ActiveCfg = Release|Any CPU
 		{52D52F3C-4F64-4957-9FE8-780B7D7F75C3}.Release|x86.Build.0 = Release|Any CPU
 		{B4689B2B-C46B-40BF-A36B-D5D837352E8B}.Debug|Any CPU.ActiveCfg = Debug|x86
+		{B4689B2B-C46B-40BF-A36B-D5D837352E8B}.Debug|Any CPU.Build.0 = Debug|x86
+		{B4689B2B-C46B-40BF-A36B-D5D837352E8B}.Debug|Any CPU.Deploy.0 = Debug|x86
```
Comment (dotMorten) [inline src/WindowsStateTriggers.sln:51]: Undo changes to the solution.

File: src/WindowsStateTriggers/AdaptiveTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                   MinWindowHeight < MaxWindowHeight;
+      }
+    }
+
+    private void OnWindowWidthPropertyChanged()
+    {
+      var window = CoreApplication.GetCurrentView()?.CoreWindow;
+      if (window != null)
+      {
+        IsActive = window.Bounds.Width >= MinWindowWidth &&
```
Comment (dotMorten) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:61]: This statement is ignoring min height.
Looks like you have the correct code in OnCoreWindowOnSizeChanged method. Why not just put that expression in its own method, and call that from all 3 event handlers?

File: src/WindowsStateTriggers/AdaptiveTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+                 size.Height < MaxWindowHeight && size.Width < MaxWindowWidth &&
+                 MinWindowHeight < MaxWindowHeight && MinWindowWidth < MaxWindowWidth;
+    }
+
+    private void OnWindowHeightPropertyChanged()
+    {
+      var window = CoreApplication.GetCurrentView()?.CoreWindow;
+      if (window != null)
+      {
+        IsActive = window.Bounds.Height >= MinWindowHeight &&
```
Comment (dotMorten) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:50]: This statement is ignoring min width

File: src/WindowsStateTriggers/AdaptiveTrigger.cs [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+    #endregion
+
+
+    #region MinWindowWidth
+
+    /// <summary>
+    /// MinWindowWidth Property name
+    /// </summary>
+    public const string MinWindowWidthPropertyName = "MinWindowWidth";
```
Comment (dotMorten) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:146]: Shouldn't be public

File: src/WindowsStateTriggers/AdaptiveTrigger.cs [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+    /// </summary>
+    public static readonly double MaxSize = int.MaxValue;
+
+
+    #region MinWindowHeight
+
+    /// <summary>
+    /// MinWindowHeight Property name
+    /// </summary>
+    public const string MinWindowHeightPropertyName = "MinWindowHeight";
```
Comment (dotMorten) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:107]: Shouldn't be public

File: src/WindowsStateTriggers/AdaptiveTrigger.cs [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+    /// <summary>
+    /// Occurs when the <see cref="IsActive" /> property has changed.
+    /// </summary>
+    public event EventHandler IsActiveChanged;
+
+
+    /// <summary>
+    /// Max size for a screen
+    /// </summary>
+    public static readonly double MaxSize = int.MaxValue;
```
Comment (dotMorten) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:99]: Shouldn't be public. Also I don't see why you need this - int.MaxValue is a constant that can be used already (and also no UI element supports this size - I believe the max size is actually 32762 last I checked)

**Comment by dotMorten** [comment]
@LocalJoost See code comments. Also how does this differ from the other adaptive trigger that was submitted?

File: src/TestApp/MainPage.xaml (hunk 1 above)
Comment (LocalJoost) [inline src/TestApp/MainPage.xaml:10]: You want to keep the inconsistent indentation? Your paryt

File: src/WindowsStateTriggers/AdaptiveTrigger.cs (hunk 3 above)
Comment (LocalJoost) [inline src/WindowsStateTriggers/AdaptiveTrigger.cs:61]: Fair enough

File: src/TestApp/MainPage.xaml (hunk 1 above)
Comment (dotMorten) [inline src/TestApp/MainPage.xaml:10]: I want to introduce changes when needed. IMHO a PR should focus on the issue, and not fix/change other stuff ;-)

**Pull request closed without merging**